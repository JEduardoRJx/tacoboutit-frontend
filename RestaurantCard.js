import React, { Component } from 'react'
import { Text, View, Image, Button, StyleSheet, Dimensions } from 'react-native';
import { Card, CardItem } from 'native-base';

export default class RestaurantCard extends Component {
  render() {
    const { name, address, city, state, zipcode, isClosed, distance, img } = this.props;
    const status = isClosed ? 'Closed' : 'Open';
    const color = isClosed ? 'red': 'green';
    return (
      <Card style={styles.cardContainer}>
         <Image
            style={styles.img}
            source={{ uri: img }}
          />
        <View style={styles.infoContainer}>
          <Text style={styles.header}>{name}</Text>
          <View style={styles.info}>
            <Text style={styles.text}>{address}</Text>
            <Text style={styles.text}>{city}, {state} {zipcode}</Text>
          </View>
          <View style={styles.statusBubble}>
            <Text style={{ color, textAlign: 'center', fontSize: 16 }}>{status}</Text>
          </View>
          <View style={styles.row}>
            <Text style={styles.text}>{distance} miles</Text>
            <Text style={styles.text}>Tap To View</Text>
          </View>
        </View>
      </Card>
    )
  }
}

const styles = StyleSheet.create({
  cardContainer: {
    backgroundColor: 'rgba(225, 225, 225, 0.3)',
    flexDirection: 'row',
    height: Dimensions.get('window').height * .25,
    width: Dimensions.get('window').width * .9,
    borderRadius: 20,
    elevation: 0,
    overflow: 'hidden',
  },
  img: { width: Dimensions.get('window').height * .25, height: Dimensions.get('window').height * .25, flex: 2 },
  infoContainer: {
    flex: 3,
    justifyContent: 'space-evenly',
  },
  header: {
    color: 'white',
    fontSize: 22,
    textAlign: 'left',
    width: '100%',
    paddingLeft: 15,
  },
  row: {
    flexDirection: 'row',
    width: '100%',
    justifyContent: 'space-evenly',
  },
  info: {
    textAlign: 'left',
    paddingLeft: 15,
    width: '100%',
  },
  text: {
    color: 'white',
    fontSize: 16,
  },
  statusBubble: {
    width: 70,
    height: 30,
    backgroundColor: 'white',
    borderRadius: 20,
    justifyContent: 'center',
    alignSelf: 'flex-end',
    marginRight: 20,
  }
});