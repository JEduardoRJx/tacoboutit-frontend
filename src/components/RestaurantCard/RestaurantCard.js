import React, { Component } from 'react'
import { Text, View, Image, Button, StyleSheet, Dimensions, TouchableOpacity } from 'react-native';
import { Card, CardItem } from 'native-base';

export default class RestaurantCard extends Component {
  render() {
    const { name, address, isClosed, img, handlePress } = this.props;
    const distance = Number(this.props.distance).toFixed(2);
    const status = isClosed ? 'Closed' : 'Open';
    const color = isClosed ? 'red': 'green';
    return (
      <TouchableOpacity style={styles.cardContainer} onPress={() => handlePress(this.props.id)}>
         <Image
            style={styles.img}
            source={{ uri: img }}
          />
        <View style={styles.infoContainer}>
          <Text style={styles.header}>{name}</Text>
          <View style={styles.info}>
            <Text style={styles.text}>{address}</Text>
          </View>
          <View style={styles.statusBubble}>
            <Text style={{ color, textAlign: 'center', fontSize: 16 }}>{status}</Text>
          </View>
          <View style={styles.row}>
            <Text style={styles.text}>{distance} miles</Text>
            <Text style={styles.text}>Tap To View</Text>
          </View>
        </View>
      </TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
  cardContainer: {
    backgroundColor: 'rgba(225, 225, 225, 0.3)',
    flexDirection: 'row',
    height: Dimensions.get('window').height * .25,
    width: Dimensions.get('window').width * .9,
    borderRadius: 20,
    elevation: 0,
    overflow: 'hidden',
    marginVertical: 10,
  },
  img: { width: Dimensions.get('window').height * .25, height: Dimensions.get('window').height * .25, flex: 2 },
  infoContainer: {
    flex: 3,
    justifyContent: 'space-evenly',
  },
  header: {
    color: 'white',
    fontSize: 22,
    textAlign: 'left',
    width: '100%',
    paddingLeft: 15,
  },
  row: {
    flexDirection: 'row',
    width: '100%',
    justifyContent: 'space-evenly',
  },
  info: {
    textAlign: 'left',
    paddingLeft: 15,
    width: '100%',
  },
  text: {
    color: 'white',
    fontSize: 16,
  },
  statusBubble: {
    width: 70,
    height: 30,
    backgroundColor: 'white',
    borderRadius: 20,
    justifyContent: 'center',
    alignSelf: 'flex-end',
    marginRight: 20,
  }
});