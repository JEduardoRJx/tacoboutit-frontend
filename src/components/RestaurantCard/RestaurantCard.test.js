import React from 'react';
import renderer from 'react-test-renderer';

import RestaurantCard from '../RestaurantCard/RestaurantCard';

describe('RestaurantCard', () => {
  it('renders correctly', () => {
    const wrapper = renderer.create(<RestaurantCard />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});