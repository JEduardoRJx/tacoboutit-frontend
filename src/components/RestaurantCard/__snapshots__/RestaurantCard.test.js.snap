// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantCard renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(225, 225, 225, 0.3)",
      "borderRadius": 20,
      "elevation": 0,
      "flexDirection": "row",
      "height": 333.5,
      "marginVertical": 10,
      "opacity": 1,
      "overflow": "hidden",
      "width": 675,
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "flex": 2,
        "height": 333.5,
        "width": 333.5,
      }
    }
  />
  <View
    style={
      Object {
        "flex": 3,
        "justifyContent": "space-evenly",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 22,
          "paddingLeft": 15,
          "textAlign": "left",
          "width": "100%",
        }
      }
    />
    <View
      style={
        Object {
          "paddingLeft": 15,
          "textAlign": "left",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "flex-end",
          "backgroundColor": "white",
          "borderRadius": 20,
          "height": 30,
          "justifyContent": "center",
          "marginRight": 20,
          "width": 70,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "green",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Open
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
          }
        }
      >
        NaN
         miles
      </Text>
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
          }
        }
      >
        Tap To View
      </Text>
    </View>
  </View>
</View>
`;
